references/classification/utils.py

```diff
@@ -161,6 +161,25 @@ def log_every(self, iterable, print_freq, header=None):
         print('{} Total time: {}'.format(header, total_time_str))
 
 
+class ExponentialMovingAverage(torch.optim.swa_utils.AveragedModel):
+    """Maintains moving averages of model parameters using an
+    exponential decay.
+    `ema_avg = decay * avg_model_param + (1 - decay) * model_param`
+    `torch.optim.swa_utils.AveragedModel <https://pytorch.org/docs/stable/optim.html#custom-averaging-strategies>`
+    is used to compute the EMA.
+    """
+    def __init__(self, model, decay, device='cpu', name='ExponentialMovingAverage'):
+        ema_avg = (lambda avg_model_param, model_param, num_averaged:
+                   decay * avg_model_param + (1 - decay) * model_param)
+        super().__init__(model, device, ema_avg)
+        self._name = name
+
+    @property
+    def name(self):
```

Comment:
I'm just wondering, what is the reason for having a `name` attribute, and why do we need it to be a `property`?
IDK if this is relevant but we can access an object's class name with `obj.__class__.__name__`

Reply:
Yeah, I think it's unnecessary, thanks!